app/src/main/java/com/teamfillin/fillin/presentation/CustomAddCompleteDialog.kt

```diff
@@ -0,0 +1,39 @@
+package com.teamfillin.fillin.presentation
+
+import android.app.Dialog
+import android.content.Context
+import android.content.DialogInterface
+import android.view.WindowManager
+import android.widget.ImageButton
+import com.teamfillin.fillin.R
+
+class CustomAddCompleteDialog(context: Context) {
+
+    private val dialog= Dialog(context)
+    private lateinit var onClickListener: OnDialogClickListener
+
+    fun setOnClickListener(listener:OnDialogClickListener){
+        onClickListener=listener
+    }
+
+    fun showDialog(){
+        dialog.setContentView(R.layout.dialog_addphoto_complete)
+        dialog.window!!.setLayout(WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.WRAP_CONTENT)
+        dialog.setCanceledOnTouchOutside(true)
+        dialog.setCancelable(true)
+        dialog.show()
+
+        val btnExit=dialog.findViewById<ImageButton>(R.id.btn_exit)
+
+
+        btnExit.setOnClickListener {
+            dialog.dismiss()
+        }
+    }
+
+    interface OnDialogClickListener{
+        fun onClicked(){
+
+        }
+    }
```

Comment:
```suggestion
    fun interface OnDialogClickListener{
        fun onClicked()
    }
```

코틀린에서 함수가 한 개인 인터페이스는 함수형 interface(fun interface)로 설계가 가능합니다. 이걸 사용하면 사용부에서 object : OnDialogClickListener 이런 구현코드를 지울 수 있음!

Reply:
아하 좋은거 배워갑니다잉,,🙂